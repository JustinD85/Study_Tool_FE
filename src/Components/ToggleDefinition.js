import React, { Component } from 'react';

class ToggleDefinition extends Component {

  render() {
    const hasSelectedWord = this.props.currentDefinition;
    return (
      <section className="ToggleDefinition">
        {hasSelectedWord &&
          <section className="word">
          <h3 >Current Word:</h3>
          <p> {this.props.currentWord}</p>
          </section>}
        <section className="definition">
          {hasSelectedWord && <p>{this.props.currentDefinition} </p>}

          {!hasSelectedWord &&
            <p className="definition-info">
            Select Words from the Word List to see it's definition.
              <br />
              Use the Filter to change the Word List Topic.
            </p>}
        </section>
      </section>
    )
  }
}


export default ToggleDefinition;