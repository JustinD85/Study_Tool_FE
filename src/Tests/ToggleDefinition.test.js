import React from 'react';
import ToggleDefinition from '../Components/ToggleDefinition';
import { shallow } from 'enzyme';

describe('<ToggleDefinition/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ToggleDefinition />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});