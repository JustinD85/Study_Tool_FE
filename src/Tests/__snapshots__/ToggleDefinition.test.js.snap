// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleDefinition/> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToggleDefinition />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <section
          className="definition"
        >
          <p
            className="definition-info"
          >
            Select Words from the Word List to see it's definition.
            <br />
            Use the Filter to change the Word List Topic.
          </p>
        </section>,
      ],
      "className": "ToggleDefinition",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <p
              className="definition-info"
            >
              Select Words from the Word List to see it's definition.
              <br />
              Use the Filter to change the Word List Topic.
            </p>,
          ],
          "className": "definition",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Select Words from the Word List to see it's definition.",
                <br />,
                "Use the Filter to change the Word List Topic.",
              ],
              "className": "definition-info",
            },
            "ref": null,
            "rendered": Array [
              "Select Words from the Word List to see it's definition.",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "Use the Filter to change the Word List Topic.",
            ],
            "type": "p",
          },
        ],
        "type": "section",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <section
            className="definition"
          >
            <p
              className="definition-info"
            >
              Select Words from the Word List to see it's definition.
              <br />
              Use the Filter to change the Word List Topic.
            </p>
          </section>,
        ],
        "className": "ToggleDefinition",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <p
                className="definition-info"
              >
                Select Words from the Word List to see it's definition.
                <br />
                Use the Filter to change the Word List Topic.
              </p>,
            ],
            "className": "definition",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Select Words from the Word List to see it's definition.",
                  <br />,
                  "Use the Filter to change the Word List Topic.",
                ],
                "className": "definition-info",
              },
              "ref": null,
              "rendered": Array [
                "Select Words from the Word List to see it's definition.",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "Use the Filter to change the Word List Topic.",
              ],
              "type": "p",
            },
          ],
          "type": "section",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
